File: CZECHMATE/CZECHMATE/Services/OpeningECO.swift
//
//  OpeningECO.swift
//  Jednoduchý offline přehled zahájení podle prefixu FEN (startovní pozice).
//

import Foundation

enum OpeningECO {
    /// Vrací název zahájení nebo nil — pouze několik běžných vzorů.
    static func title(forFEN fen: String) -> String? {
        let t = fen.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
        let parts = t.split(separator: " ")
        guard let first = parts.first else { return nil }
        let board = String(first)

        if board == "rnbqkbnr/pppppppp/8/8/8/8/pppppppp/rnbqkbnr" {
            return "Základní pozice"
        }
        if board.hasPrefix("rnbqkbnr/pppppppp/8/8/4p3/8/pppp1ppp/rnbqkbnr") {
            return "Obrana Englund (neobvyklá)"
        }
        if board.hasPrefix("rnbqkbnr/pppppppp/8/8/3p4/8/ppp2ppp/rnbqkbnr") {
            return "Scandinávská obrana"
        }
        if board.hasPrefix("rnbqkbnr/pppp1ppp/8/4p3/4p3/8/pppp1ppp/rnbqkbnr") {
            return "Otevřená hra"
        }
        return nil
    }
}

File: flutter_czechmate/lib/core/utils/opening_eco.dart
/// Parita `OpeningECO.swift` — jednoduchý offline název zahájení podle prefixu FEN.
class OpeningEco {
  OpeningEco._();

  /// Vrací název zahájení nebo `null` — pouze několik běžných vzorů.
  static String? titleForFen(String fen) {
    final t = fen.trim().toLowerCase();
    final parts = t.split(RegExp(r'\s+'));
    if (parts.isEmpty) return null;
    final board = parts.first;

    if (board == 'rnbqkbnr/pppppppp/8/8/8/8/pppppppp/rnbqkbnr') {
      return 'Starting position';
    }
    if (board.startsWith('rnbqkbnr/pppppppp/8/8/4p3/8/pppp1ppp/rnbqkbnr')) {
      return 'Englund Defence (unusual)';
    }
    if (board.startsWith('rnbqkbnr/pppppppp/8/8/3p4/8/ppp2ppp/rnbqkbnr')) {
      return 'Scandinavian Defence';
    }
    if (board.startsWith('rnbqkbnr/pppp1ppp/8/4p3/4p3/8/pppp1ppp/rnbqkbnr')) {
      return 'Open game';
    }
    return null;
  }
}
